kotlin_engine/src/main/java/com/github/paylike/kotlin_engine/viewmodel/PaylikeEngine.kt

```diff
@@ -0,0 +1,287 @@
+package com.github.paylike.kotlin_engine.viewmodel
+
+import android.util.Log
+import com.github.paylike.kotlin_client.PaylikeClient
+import com.github.paylike.kotlin_client.domain.dto.payment.request.PaymentData
+import com.github.paylike.kotlin_client.domain.dto.payment.request.card.ExpiryDto
+import com.github.paylike.kotlin_client.domain.dto.payment.request.card.PaylikeCardDto
+import com.github.paylike.kotlin_client.domain.dto.payment.request.integration.PaymentIntegrationDto
+import com.github.paylike.kotlin_client.domain.dto.payment.request.test.PaymentTestDto
+import com.github.paylike.kotlin_client.domain.dto.payment.response.PaylikeClientResponse
+import com.github.paylike.kotlin_client.domain.dto.tokenize.request.TokenizeData
+import com.github.paylike.kotlin_client.domain.dto.tokenize.request.TokenizeTypes
+import com.github.paylike.kotlin_engine.helper.PaylikeEngineError
+import com.github.paylike.kotlin_engine.helper.exceptions.*
+import com.github.paylike.kotlin_engine.model.repository.EngineRepository
+import com.github.paylike.kotlin_engine.model.service.ApiMode
+import com.github.paylike.kotlin_luhn.PaylikeLuhn
+import com.github.paylike.kotlin_money.PaymentAmount
+import com.github.paylike.kotlin_request.exceptions.PaylikeException
+import java.util.*
+import java.util.function.Consumer
+import kotlin.reflect.full.superclasses
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+
+/** Paylike engine */
+// TODO KDoc of engine
+class PaylikeEngine(private val merchantId: String, private val apiMode: ApiMode) : Observable() {
+
+    val repository: EngineRepository = EngineRepository()
+
+    var currentState: EngineState = EngineState.WAITING_FOR_INPUT
+        private set
+
+    var error: PaylikeEngineError? = null
+        private set
+
+    private val apiService: PaylikeClient = PaylikeClient()
```

Comment:
`apiClient` is a bit better name here. By `apiService` I directly think about a service that can be found somewhere in this repository. So I rather think about an internal service layer rather than a given third party service for example